ansible_builder/containerfile.py

```diff
@@ -83,20 +83,32 @@ def prepare(self):
 
         ######################################################################
         # First stage (aka, galaxy): install roles/collections
+        #
+        # NOTE: This stage is skipped if there are no galaxy requirements.
+        #
+        # We first insert another intermediary stage that is used only to
+        # validate that the ansible-galaxy command is available in the base
+        # image. Failure to find the command should cause a non-zero exit and
+        # fail the build.
         ######################################################################
 
-        self.steps.extend([
-            "",
-            "# Galaxy build stage",
-            "FROM base as galaxy",
-        ])
+        if self.definition.get_dep_abs_path('galaxy'):
+            self.steps.extend([
+                "",
+                "# Validate that we have an ansible-galaxy command available"
+                "FROM base as check_galaxy",
```

Comment:
I think I'm the one that suggested making the Galaxy check its own build stage, but now I'm second-guessing it... Since the Galaxy stage will never be persisted, we don't care about the extra layers that the check command will create on it, so maybe the check script invocation should just be inline with the main Galaxy stage (question is if the prepend steps go before or after the check- I could see reasons for either).